### PR TITLE
Metamodel haystack

### PR DESCRIPTION
**Pull request by sachintendulkar576123**
## Proposed changes

This PR:

1. Removed the haystack specific code from wrap_pipeline.py , wrap_openai.py and wrap_node.py and consolidated under wrap_common.py
2.  Attributes of inference span is captured
3. Vector store properties and context input and output are being captured.

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/monocle_apptrace/wrap_common.py [hunk 1]
```diff
@@ -75,7 +76,17 @@ def get_embedding_model_for_vectorstore(instance):
         'type': VECTOR_STORE,
     },
 }
+def get_embedding_model_haystack(instance):
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/wrap_common.py:79]: Can this be moved to metamodel?

**Comment by kshitiz-okahu** [comment]
Can we remove haystack folder which has wrap_node.py and wrap_openai.py, since the code is not being used now.

File: src/monocle_apptrace/wrap_common.py [hunk 2]
```diff
@@ -323,6 +346,17 @@ def get_input_from_args(chain_args):
...
+
+    if (response is not None and isinstance(response, dict) and "meta" in response):  # haystack
+        token_usage = response["meta"][0]["usage"]
+        meta_dict={}
+        temperature = instance.__dict__.get("temperature", None)
+        meta_dict.update({"temperature": temperature})
+        meta_dict.update({"completion_tokens": token_usage.get("completion_tokens")})
+        meta_dict.update({"prompt_tokens": token_usage.get("prompt_tokens")})
+        meta_dict.update({"total_tokens": token_usage.get("total_tokens")})
+        span.add_event(META_DATA, meta_dict)
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/wrap_common.py:358]: This code looks to be an exact copy of code below

File: src/monocle_apptrace/wrap_common.py [hunk 3]
```diff
@@ -90,7 +101,15 @@ def task_wrapper(tracer: Tracer, to_wrap, wrapped, instance, args, kwargs):
     elif to_wrap.get("span_name"):
         name = to_wrap.get("span_name")
     else:
-        name = f"langchain.task.{instance.__class__.__name__}"
+        name = get_fully_qualified_class_name(instance)
+
+    if 'haystack.core.pipeline.pipeline' in to_wrap['package']:
+        embedding_model = get_embedding_model_haystack(instance)
+        set_embedding_model(embedding_model)
+        inputs = set()
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/wrap_common.py:109]: What does this do?

**Review by kshitiz-okahu — CHANGES_REQUESTED** [review]

File: src/monocle_apptrace/wrap_common.py (hunk 3 above)
Comment (sachintendulkar576123) [reply src/monocle_apptrace/wrap_common.py:109]: we are getting embedding model in pipeline package , but we need embedding in retriever package , thats why we are getting and setting the embedding model in pipeline package to utilize in retriever package

File: src/monocle_apptrace/wrap_common.py (hunk 2 above)
Comment (sachintendulkar576123) [reply src/monocle_apptrace/wrap_common.py:358]: the way meta_dict for tokens is same , but reading of token_usage and condition is different .

File: src/monocle_apptrace/wrap_common.py (hunk 1 above)
Comment (sachintendulkar576123) [reply src/monocle_apptrace/wrap_common.py:79]: embedding model in coming in pipeline package and we need it in retriever package , thats why we cannot remove this code

File: src/monocle_apptrace/wrap_common.py (hunk 2 above)
Comment (kshitiz-okahu) [reply src/monocle_apptrace/wrap_common.py:358]: Can we move the common code like this

token_usage = None
if (response is not None and isinstance(response, dict) and "meta" in response):  # haystack
    token_usage = response["meta"][0]["usage"]

# extract token uasge from langchain openai
if (response is not None and hasattr(response, "response_metadata")):
    response_metadata = response.response_metadata
    token_usage = response_metadata.get("token_usage")

meta_dict = {}
if token_usage is not None:
    temperature = instance.__dict__.get("temperature", None)
    meta_dict.update({"temperature": temperature})
    meta_dict.update({"completion_tokens": token_usage.get("completion_tokens")})
    meta_dict.update({"prompt_tokens": token_usage.get("prompt_tokens")})
    meta_dict.update({"total_tokens": token_usage.get("total_tokens")})
    span.add_event(META_DATA, meta_dict)

**Review by oi-raanne — APPROVED** [review]

**Review by kshitiz-okahu — APPROVED** [review]

**Pull request merged**